lib/stub_ui/lib/src/ui/window.dart

```diff
@@ -617,6 +618,97 @@ class Window {
   /// intrusions in the display (e.g. overscan regions on television screens or
   /// phone sensor housings).
   ///
+  /// Unlike [Window.padding], this value does not change relative to
+  /// [Window.viewInsets]. For example, on an iPhone X, it will not change in
+  /// response to the soft keyboard being visible or hidden, whereas
+  /// [Window.padding] will.
+  ///
+  /// For example, on an iPhone X when the keyboard is not showing, the
+  /// viewPadding and the padding will be equal, as there are no insets at play.
+  /// In this scenario, the [Window.viewInsets.bottom] will be 0, and the
+  /// [Window.padding.bottom] and [Window.viewPadding.bottom] will be some
+  /// positive number specified by the OS:
+  ///
+  /// ```
+  ///   ▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
+  ///  ╱     ████████     ╲
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃
+  /// ┃                    ┃  _____________________________________
+  /// ┃      ━━━━━━━━      ┃  viewPadding.bottom == padding.bottom
+  ///  ╲▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁╱   _____________________________________
+  /// ```
+  ///
+  /// Whereas when the keyboard is showing, the viewPadding and padding will be
```

Comment:
nit: starting with 'Whereas' seems wrong since there's no comparison within the sentence.